canonical-web-and-design/maas-ui
# Refactor scriptresult [WIP]

## Done
- `scriptresult` now contains a bare list of scriptResults with no associated machineId. Confusingly the websocket API refers to these as `noderesults`, but the objects returned are in fact `ScriptResult`s.
- `nodescriptresult` is a new root model which is essentially a lookup/join table, mapping machine ids to scriptResult ids. This approach is recommended in the redux toolkit docs under [normalizing state shape](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape).

### Note
This branch does not implement support for updates, which will come in a follow-up branch.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit the machine testing tab and ensure script results still render as expected.

## Fixes

Fixes: #1983
